Sources/TensorFlow/Initializers.swift

```diff
@@ -509,17 +531,28 @@ public extension Tensor where Scalar: BinaryFloatingPoint,
     }
 }
 
+//===------------------------------------------------------------------------------------------===//
+// Variance Scaling
+//===------------------------------------------------------------------------------------------===//
+
 fileprivate extension Tensor where Scalar: TensorFlowFloatingPoint {
-    private static func glorot(
-        fromStandardUniform randomUniform: __shared Tensor<Scalar>,
-        shape: __shared TensorShape
-    ) -> Tensor<Scalar> {
-        let spatialDimCount = shape.count - 2
-        let receptiveField = shape[0..<spatialDimCount].contiguousSize
-        let fanIn = shape[shape.count - 2] * receptiveField
-        let fanOut = shape[shape.count - 1] * receptiveField
-        let minusOneToOne = 2 * randomUniform - 1
-        return Scalar.sqrt(Scalar(6) / Scalar(fanIn + fanOut)) * minusOneToOne
+    // Returns the input and output channel counts, `(fanIn, fanOut)`, of a given tensor shape.
+    private static func fans(shape: TensorShape) -> (in: Int, out: Int) {
+        precondition(shape.count > 1,
+            "Fans cannot be computed for tensors with fewer than 2 dimensions. Got: \(shape.count)")
+
+        // Fans for a 2-D tensor, e.g. `Dense`/`Embedding` weights.
+        if shape.count == 2 {
+            return (shape[0], shape[1])
+        }
+        // Fans for tensors with rank greater than `2`, specifically convolution filters.
+        let lastSpatialDimension = shape.count - 3
```

Comment:
Better call this "axis" rather than "dimension" to be consistent with naming in the rest of the library.